apps/opik-backend/src/main/java/com/comet/opik/api/resources/v1/events/BaseRedisSubscriber.java

```diff
@@ -470,8 +539,55 @@ private Optional<Long> extractTimeFromMessageId(StreamMessageId messageId) {
         return Optional.empty();
     }
 
+    /**
+     * Non-retryable exceptions are programming errors or validation failures that won't succeed on retry.
+     * All other exceptions are considered retryable (transient errors like network issues, timeouts, etc.)
+     * Unknown exceptions default to retryable for safety.
+     */
+    private boolean isRetryableException(Throwable exception) {
+        return !NON_RETRYABLE_EXCEPTIONS.contains(exception.getClass());
+    }
+
+    /**
+     * The delivery count indicates how many times a message has been delivered to consumers.
+     *
+     * @param messageId the message ID to query
+     * @return Mono with the delivery count (0 if not found or on error)
+     */
+    private Mono<Long> getDeliveryCount(StreamMessageId messageId) {
+        return listPending(messageId)
+                .map(PendingEntry::getLastTimeDelivered)
```

Comment:
Minor: It might be worth to add a comment here clarifying the `PendingEntry::getLastTimeDelivered` actually returns the delivery count. I went reading the [Redisson javadoc](https://javadoc.io/doc/org.redisson/redisson/3.8.2/org/redisson/api/PendingEntry.html) since I thought you mistakenly used the timestamp :)